IliasDownloaderUniMA/__init__.py

```diff
@@ -100,6 +120,29 @@ def login(self, login_id, login_pw):
 			raise ConnectionError("Couldn't log into ILIAS. Make sure your provided uni-id and the password are correct.")
 
 
+	def extractUserCourses(self):
+		"""
+		Extracts the users subscribed courses from the login_soup
+
+		:returns:   List of dicts, with keys name, url and iliasid
+		:rtype:     list
+		"""
+
+		courses = []
+
+		containers = self.login_soup.select("div.il_ContainerListItem")
+		for container in containers:
+			title_elem = container.select("a.il_ContainerItemTitle")[0]
+
+			course_name = self.cleanCourseName(title_elem.string)
+			iliasid = self.extractIdFromUrl(title_elem["href"])
+
+			url = self.createIliasUrl(iliasid)
+			courses += [{"name": course_name, "url": url, "iliasid": iliasid}]
+
+		return courses
```

Comment:
I'd prefer the approach with the two methods. I think the tasks are logically distinct and will be a lot easier to write tests for this way.
See my design philosophy at the bottom if you're interested why I coded it this way.

Reply:
>     * It adds **all** courses from the user's Ilias main page. Depending on the user's Ilias settings, even courses from past semesters.
Yeah, that's very suboptimal. I should probably redo that.
>     * In my case it adds also no regular semester courses like 'License Information (Student University of Mannheim)' or 'Digitalisierung in der Lehre (Hilfskräfte und Tutorinnen und Tutoren)'. I don't see the need to scan these folders.
How can we filter them out? Do all non regular courses miss the semester number, or do we need to find another way?
> A better solution might be to scan only the current semester's courses.
How about scanning by default only the current semester, but being able to download other semesters if wanted?

Reply:
> I'd prefer the approach with the two methods. I think the tasks are logically distinct and will be a lot easier to write tests for this way.

I see your point. I never indented to write tests for this small project (probably because it's my last semester),

> How about scanning by default only the current semester, but being able to download other semesters if wanted?

👍 

> How can we filter them out? Do all non regular courses miss the semester number, or do we need to find another way?

Yeah, I think checking the `il_ContainerItemTitle` for the regex pattern `r'([A-Z]{3} \d{4})'` will do it.